include/vapor/DCP.h

```diff
@@ -0,0 +1,192 @@
+
+    //! \copydoc DC::GetCoordVarNames()
+    //!
+    virtual size_t getNumRefLevels(string varname) const { return (1); }
+
+    //! \copydoc DC::GetMapProjection()
+    //!
+    virtual string getMapProjection() const
+    {
+        // Projections not supported yet :-(
+        //
+        return (_proj4String);
+    }
+
+    //! \copydoc DC::GetAtt()
+    //!
+    virtual bool getAtt(string varname, string attname, vector<double> &values) const;
+    virtual bool getAtt(string varname, string attname, vector<long> &values) const;
+    virtual bool getAtt(string varname, string attname, string &values) const;
+
+    //! \copydoc DC::GetAttNames()
+    //!
+    virtual std::vector<string> getAttNames(string varname) const;
+
+    //! \copydoc DC::GetAttType()
+    //!
+    virtual XType getAttType(string varname, string attname) const;
+
+    //! \copydoc DC::GetDimLensAtLevel()
+    //!
+    virtual int getDimLensAtLevel(string varname, int level, std::vector<size_t> &dims_at_level, std::vector<size_t> &bs_at_level) const;
+    virtual int getDimLensAtLevel(string varname, int level, std::vector<size_t> &dims_at_level, std::vector<size_t> &bs_at_level, long ts) const;
+
+
+    //! \copydoc DC::OpenVariableRead()
+    //!
+    virtual int openVariableRead(size_t ts, string varname, int, int) { return (DCP::openVariableRead(ts, varname)); }
+
+    virtual int openVariableRead(size_t ts, string varname);
+
+
+    //! \copydoc DC::CloseVariable()
+    //!
+    virtual int closeVariable(int fd);
+
+    //! \copydoc DC::ReadRegion()
+    //
+    virtual int readRegion(int fd, const vector<size_t> &min, const vector<size_t> &max, float *region) { return (_readRegionTemplate(fd, min, max, region)); }
+    virtual int readRegion(int fd, const vector<size_t> &min, const vector<size_t> &max, int *region) { return (_readRegionTemplate(fd, min, max, region)); }
+
+    //! \copydoc DC::ReadRegionBlock()
+    //!
+    virtual int readRegionBlock(int fd, const vector<size_t> &min, const vector<size_t> &max, float *region) { return (_readRegionTemplate(fd, min, max, region)); };
+    virtual int readRegionBlock(int fd, const vector<size_t> &min, const vector<size_t> &max, int *region) { return (_readRegionTemplate(fd, min, max, region)); }
+
+    //! \copydoc DC::VariableExists()
+    //!
+    virtual bool variableExists(size_t ts, string varname, int reflevel = 0, int lod = 0) const;
+
+private:
+    NetCDFCollection *_ncdfc;
+    VAPoR::UDUnits    _udunits;
+
+    string                          _proj4String;
+    std::map<string, DC::Dimension> _dimsMap;
+    std::map<string, DC::CoordVar>  _coordVarsMap;
+    std::map<string, DC::AuxVar>    _auxVarsMap;
+    std::map<string, DC::Mesh>      _meshMap;
+    std::map<string, DC::DataVar>   _dataVarsMap;
+    std::map<string, string>        _coordVarKeys;
+    std::map<string, string>        _sanitizedToOriginalMap;
+
+    const string          nodeFaceVar = "cellsOnVertex";
```

Comment:
Please follow underscore convention for naming private variables.

Reply:
Done